### ISSUE TITLE
Can not decrypt book pdf with Adobe DRM (adobe Digital Editions 2.0.1)

### ISSUE THREAD

**Issue by plantebenoit** [apprenticeharper/DeDRM_tools#348]
Hi.

I installed Adobe Digital Editions 2.0.1 and I have no problem opening a book, but I am not able to decrypt a book using Calibre and the DeDRM plugin.
I had other version of Adobe Digital Editions installed before, but I uninstalled it, and installed 2.0.1 instead.  Still it does not work.

Can you help me?

Here are some screenshots:
ADE version
![capture1](https://user-images.githubusercontent.com/10352949/29737002-1ecbb202-89d5-11e7-847c-90369a4a5078.JPG)

Authorization on my computer (I hiden my email)
![capture2](https://user-images.githubusercontent.com/10352949/29737019-5859466a-89d5-11e7-9051-9af9fbd2ce55.JPG)

The book was imported in Adobe Digital Editions, and I can open it and read it without any problem.
![book_opened_ade](https://user-images.githubusercontent.com/10352949/29737027-6f6aa696-89d5-11e7-9bb2-1ea7744a806e.JPG)

I opened Calibre in debug mode, here is the log:

calibre Journal de débogage
calibre 2.58  embedded-python: True is64bit: False
Windows-8-6.2.9200 Windows ('32bit', 'WindowsPE')
32bit process running on 64bit windows
('Windows', '8', '6.2.9200')
Python 2.7.9
Windows: ('8', '6.2.9200', '', 'Multiprocessor Free')
Successfully initialized third party plugins: DeDRM
Starting up...
Started up in 14.98 seconds with 0 books
stdout+stderr from file dialog helper: ['', '']
piped data from file dialog helper: ['\x02-\xd7\xd2H\xdd\x07\x93C")\xd5\xa3=\x16\xc3\x08\x19\xce\x9dXj\xb1M\x1d\x01{}\xa2\xb5\xbd\x8a', 'C:\\Users\\Benoit\\Documents\\My Digital Editions']
DeDRM v6.5.5: Trying to decrypt H Mahler Seminar-style Slides for CAS Exam 8, Fall 2017.pdf
DeDRM v6.5.5: H Mahler Seminar-style Slides for CAS Exam 8, Fall 2017.pdf is a PDF ebook
DeDRM v6.5.5: Looking for new default Adobe Digital Editions Keys after 0.4 seconds
DeDRM v6.5.5: Exception when getting default Adobe Key after 0.4 seconds
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 446, in PDFDecrypt
  File "calibre_plugins.dedrm.adobekey", line 378, in adeptkeys
  File "calibre_plugins.dedrm.adobekey", line 358, in CryptUnprotectData
ADEPTError: Failed to decrypt user key key (sic)
DeDRM v6.5.5: Ultimately failed to decrypt after 0.4 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 154, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 625, in run
  File "calibre_plugins.dedrm.__init__", line 500, in PDFDecrypt
DeDRMError: DeDRM v6.5.5: Ultimately failed to decrypt after 0.4 seconds. Read the FAQs at Harper's repository: https://github.com/apprenticeharper/DeDRM_tools/blob/master/FAQs.md
Syntax Error: Couldn't find the 'EBX_HANDLER' security handler
pdfinfo errored out with return code: 1
Traceback (most recent call last):
  File "site-packages\calibre\customize\ui.py", line 376, in get_file_type_metadata
  File "site-packages\calibre\customize\builtins.py", line 323, in get_metadata
  File "site-packages\calibre\ebooks\metadata\pdf.py", line 99, in get_metadata
ValueError: Could not read info dict from PDF
Added H Mahler Seminar-style Slides for CAS Exam 8, Fall 2017 to db in: 1.6
Added 1 books in 3.1 seconds
Exception TypeError: "'NoneType' object is not callable" in <bound method NativeFunction.__del__ of <calibre_plugins.dedrm.adobekey.NativeFunction object at 0x06E152F0>> ignored
Exception TypeError: "'NoneType' object is not callable" in <bound method NativeFunction.__del__ of <calibre_plugins.dedrm.adobekey.NativeFunction object at 0x06E15410>> ignored

**Comment by ElleKayEm** [comment]
Some piece of the previous installation lurking around might be causing an issue.  I've seen people solve it by completely removing ADE including preferences and registry entries. Then reinstalling ADE 2.0.1, reauthorizing with their Adobe ID, and trying again.

**Comment by plantebenoit** [comment]
Humm how could I do that? Do you know the path for the preferences and registry?

**Comment by ElleKayEm** [comment]
People have mentioned this registry entry for the Adobe ID activation: HKEY_CURRENT_USER\Software\Adobe\Adept

But I really don't know what else might get left behind after uninstall and possibly cause an issue.   Of course, you always need to be very cautious if making changes to the registry.

**Comment by plantebenoit** [comment]
Thank you it worked!

For other people who got that issue:

I uninstalled ADE, then deleted that registry key
HKEY_CURRENT_USER\Software\Adobe\Adept

Also I went in the "AppData" folder in My Documents, then opened Local/Adobe, Local/Adobe_Incorporated and Roaming/Adobe, and I deleted all the folders there that were called either Digital Editions or Adept.

Then I reinstalled ADE 2.0.1 and the calibre plugin worked perfectly.
